### ISSUE TITLE
evaluation in dist_train.sh doesn't work. multi_gpu_test fails

### ISSUE THREAD

**Issue by Kaeseknacker** [open-mmlab/mmdetection#4207]
**Describe the bug**
I have problems with evaluation while training with dist_train.sh. 

I built the newest MMDetection version today. 
The training (and evaluation) on a single GPU works. 
test.py and dist_test.py each work too. 
The training on multiple GPUs also works (dist_train.sh), but the evaluation after one epoch fails. 
```evaluation = dict(interval=1, metric='mAP')```
Unfortunately, I don't understand the error listed below and I can't find anything about it with Google.

**Environment**

```
sys.platform: linux
Python: 3.7.9 (default, Aug 31 2020, 12:42:55) [GCC 7.3.0]
CUDA available: True
GPU 0: Quadro P6000
GPU 1,3: TITAN X (Pascal)
GPU 2: GeForce GTX 1080 Ti
CUDA_HOME: /home/spraul/lib/cuda_10.1.243_cudnn_7.6.3_linux_nvidia418.87_x64/cuda_10.1.243_cudnn_7.6.3_linux_nvidia418.87_x64
NVCC: Cuda compilation tools, release 10.1, V10.1.243
GCC: gcc (Debian 8.3.0-6) 8.3.0
PyTorch: 1.7.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON,

TorchVision: 0.8.1
OpenCV: 4.4.0
MMCV: 1.2.0
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 10.1
MMDetection: 2.6.0+fc5c286
```

**Error traceback**

```
Traceback (most recent call last):
  File "./tools/train.py", line 181, in <module>
    main()
  File "./tools/train.py", line 177, in main
    meta=meta)
  File "/home/user/mmdetection_Buster/mmdet/apis/train.py", line 150, in train_detector
    runner.run(data_loaders, cfg.workflow, cfg.total_epochs)
  File "/home/user/anaconda3/envs/mmb/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 125, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/home/user/anaconda3/envs/mmb/lib/python3.7/site-packages/mmcv/runner/epoch_based_runner.py", line 54, in train
    self.call_hook('after_train_epoch')
  File "/home/user/anaconda3/envs/mmb/lib/python3.7/site-packages/mmcv/runner/base_runner.py", line 307, in call_hook
    getattr(hook, fn_name)(self)
  File "/home/user/mmdetection_Buster/mmdet/core/evaluation/eval_hooks.py", line 237, in after_train_epoch
    gpu_collect=self.gpu_collect)
  File "/home/user/mmdetection_Buster/mmdet/apis/test.py", line 97, in multi_gpu_test
    result = model(return_loss=False, rescale=True, **data)
  File "/home/user/anaconda3/envs/mmb/lib/python3.7/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/home/user/anaconda3/envs/mmb/lib/python3.7/site-packages/torch/nn/parallel/distributed.py", line 606, in forward
    if self.reducer._rebuild_buckets():
RuntimeError: replicas_[0].size() == rebuilt_param_indices_.size() INTERNAL ASSERT FAILED at "/opt/conda/conda-bld/pytorch_1603729006826/work/torch/csrc/distributed/c10d/reducer.cpp":1326, please report a bug to PyTorch. rebuilt parameter indices size is not same as original model parameters size.93 versus 12183
```

**Comment by Co4AI** [comment]
meet the same problem. And I fix it by https://github.com/pytorch/pytorch/issues/47050 .

**Comment by Kaeseknacker** [comment]
Thanks for your reply. So the current solution is to use Pytorch 1.6?

**Comment by hellock** [comment]
https://github.com/open-mmlab/mmcv/issues/636

**Comment by hellock** [comment]
Fixed in open-mmlab/mmcv#769
